docs: document rules (#700)
asyncapi/parser-js: README.md

- `jsonPath()` which returns the JSON Path of the given object.
- `meta()` which returns the metadata of a given object, like a parsed AsyncAPI Document.

## Spectral rulesets

> Validation of AsyncAPI documents in ParserJS is powered by [Spectral](https://github.com/stoplightio/spectral). For this reason, it is possible to use own rulesets/rules, or overwrite existing ones, by passing `ruleset` option to the Parser instance:

Reviewer: ```suggestion
Validation of AsyncAPI documents in ParserJS is powered by [Spectral](https://github.com/stoplightio/spectral). For this reason, it is possible to use your own rulesets/rules, or overwrite existing ones, bypassing the `ruleset` option to the Parser instance:
```

Author: @alequetzalli Thanks!

`bypassing` means that something will be missed, not passed, yes? I want to wrote, that something need to be passed 😄

Reviewer: hehe 😜  LMGTFY 

<img width="786" alt="Screen Shot 2023-02-13 at 6 44 05 PM" src="https://user-images.githubusercontent.com/19964402/218625503-400f8b06-a516-4dcc-9f3a-6890f1f2ff06.png">

And ah, ok! got it... then let me change that edit 😄✌🏽

Author: @alequetzalli Thanks! ❤️

```ts
import { Parser, stringify, unstringify } from '@asyncapi/parser';

const parser = new Parser({
  ruleset: {
    extends: [],
    rules: {
      'asyncapi-defaultContentType': 'off',
      'asyncapi-termsOfService': {
        description: 'Info "termsOfService" should be present and non-empty string.',
        recommended: true,
        given: '$',
        then: {
          field: 'info.termsOfService',
          function: 'truthy',
        },
      },
    }
  }
});

// One of diagnostics should have `asyncapi-termsOfService` code with `warning` severity, because we didn't define the `$.info.termsOfService` field.
// Also we should won't see the diagnostics related to the omitted `defaultContentType` field.
const diagnostics = await parser.validate(`
  asyncapi: '2.0.0'
  info:
    title: Example AsyncAPI specification
    version: '0.1.0'
  channels: {}
`);
```

> ParserJS has some built-in Spectral rulesets that validate AsyncAPI documents and inform about good practices. [Checks them](./docs/ruleset) for more info.

Reviewer: ```suggestion
[ParserJS has some built-in Spectral rulesets](./docs/ruleset) that validate AsyncAPI documents and inform on good practices. 
```
